src/pip/_internal/req/req_install.py

```diff
@@ -606,6 +606,18 @@ def egg_info_path(self):
                     key=lambda x: x.count(os.path.sep) +
                     (os.path.altsep and x.count(os.path.altsep) or 0)
                 )
+                logger.warning(
+                    (
+                        'Multiple "*.egg-info/" in "%(base)s" found: %(filenames)s. '  # noqa: E501
+                        'In most cases, You should delete them except one which corresponding to the package. '  # noqa: E501
+                        'pip is going to refer %(chosen)s to install the package.'  # noqa: E501
+                    ),
+                    {
```

Comment:
Since there are only 3 parameters, I think it would be fine to just use `%s` in the log message and pass them as separate arguments.